src/main/java/com/mongodb/kafka/connect/sink/SinkConfigSoftValidator.java

```diff
@@ -32,58 +32,43 @@
 import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.KEY_PROJECTION_TYPE_CONFIG;
 import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.KEY_PROJECTION_TYPE_DEFAULT;
 import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.POST_PROCESSOR_CHAIN_CONFIG;
-import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.TOPIC_OVERRIDE_PREFIX;
 import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.VALUE_PROJECTION_LIST_CONFIG;
 import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.VALUE_PROJECTION_LIST_DEFAULT;
 import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.VALUE_PROJECTION_TYPE_CONFIG;
 import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.VALUE_PROJECTION_TYPE_DEFAULT;
 import static com.mongodb.kafka.connect.sink.MongoSinkTopicConfig.WRITEMODEL_STRATEGY_CONFIG;
-import static com.mongodb.kafka.connect.util.Assertions.assertNotNull;
-import static com.mongodb.kafka.connect.util.Assertions.assertTrue;
-import static com.mongodb.kafka.connect.util.VisibleForTesting.AccessModifier.PRIVATE;
+import static java.lang.String.format;
+import static java.util.Collections.unmodifiableSet;
+import static java.util.stream.Collectors.toSet;
 
-import java.util.AbstractMap;
-import java.util.Arrays;
 import java.util.Collection;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.HashSet;
 import java.util.Map;
-import java.util.Map.Entry;
 import java.util.Set;
 import java.util.function.Consumer;
-import java.util.function.Function;
-import java.util.stream.Collectors;
-
-import org.apache.kafka.common.config.ConfigDef;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import com.mongodb.annotations.Immutable;
-import com.mongodb.lang.Nullable;
-
-import com.mongodb.kafka.connect.util.VisibleForTesting;
-
-/** Logs information about potential configuration issues. */
-public final class SinkConfigSoftValidator {
```

Comment:
This class was split into `ConfigSoftValidator` (contains the validation functionality) and `SinkConfigSoftValidator`/`SourceConfigSoftValidator` that contain specific business rules to check (unused, deprecated, conflicting properties).

Reply:
Ack